inc/authldap.class.php

```diff
@@ -2550,6 +2577,21 @@ static function ldapImportUserByServerId(array $params, $action, $ldap_server,
                //Get information from LDAP
                if ($user->getFromLDAP($ds, $config_ldap->fields, $user_dn, addslashes($login),
                                     ($action == self::ACTION_IMPORT))) {
+                  //Get the ID by sync field (Used to check if restoration is needed)
+                  if (!($tmp_users_id = User::getIdByfield($params['user_field'], $login))) {
+                     //In case user id as changed : get id by dn (Used to check if restoration is needed)
+                     $tmp_users_id = User::getIdByfield('user_dn', $user_dn);
```

Comment:
you can use $user->getFromDBbySyncField & $user->getFromDBbyDn for the above lines